src/hooks/useMutagenNfts.ts

```diff
@@ -18,52 +20,45 @@ const getMutagenNfts = async (
     call: ["tokenOfOwnerByIndex(address,uint256)(uint256)", account, i],
     returns: [[i, (val) => parseInt(val)]],
   }))
-  console.log(idRequests)
-
   const {
-    results: { transformedIds },
+    results: { transformed: idResults },
   } = await aggregate(idRequests, config)
-  console.log(transformedIds)
-  const ids = Object.values(transformedIds)
+  const ids: number[] = Object.values(idResults)
 
   const dataRequests = ids.map((id) => ({
     target: mutagenAddress,
     call: ["unpackPrintId(uint256)(uint8,uint256,uint16)", id],
     returns: [[id], [], []],
   }))
   const {
-    results: { transformedData },
+    results: { transformed: dataResults },
   } = await aggregate(dataRequests, config)
-  const correctIdsArray = Object.entries(transformedData).slice(0, -1)
-  const data = correctIdsArray.reduce((arr, [tokenId, genesisIdx]) => {
-    // eslint-disable-next-line no-bitwise
-    const type = tokenId as any & 3
-    if (type === 0) {
-      arr.push(`Genesis_${genesisIdx}`)
-    } else if (type === 1) {
-      arr.push(`Print_${genesisIdx}`)
-    }
-    return arr
-  }, [])
-  return data
+
+  return Object.entries(dataResults)
+    .slice(0, -1)
+    .map(([tokenId, genesisIdx]) => {
+      // eslint-disable-next-line no-bitwise
+      const type = parseInt(tokenId) & 3
+
+      if (type === 0) return `Genesis_${genesisIdx}`
+      if (type === 1) return `Print_${genesisIdx}`
+      return null
+    })
 }
 
 const useMutagenNfts = (requirementType: RequirementType, token: Token) => {
   const { account } = useWeb3React()
-  const amount = useBalance(token)
+  const amount: any = useBalance(token)
 
   const shouldFetch = requirementType === "NFT_HOLD" && !!token && amount > 0
 
-  /**
-   * TODO: deduping doesn't work for some reason. getMutagenNfts get's called 4 times
-   * in a row even though the parameters haven't changed
-   */
-  const { data, mutate } = useSWR(
-    shouldFetch ? ["mutagen", amount, token.address, account] : null,
+  const { data } = useSWR(
+    shouldFetch ? ["mutagen", parseInt(amount), token.address, account] : null,
     getMutagenNfts
   )
 
-  useKeepSWRDataLiveAsBlocksArrive(mutate)
+  // can't use it because deduping doesn't work with mutate and this way it triggers too many fetches
```

Comment:
Shouldn't we just put `{ refreshInterval: 10_000 }` in every SWR call, we would use `useKeepSWRDataLiveAsBlocksArrive` on?